consensus/polybft/bridge_batch.go

```diff
@@ -0,0 +1,172 @@
+package polybft
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/bls"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/types"
+	merkle "github.com/Ethernal-Tech/merkle-tree"
+)
+
+// PendingBridgeBatch holds merkle trie of bridge transactions accompanied by epoch number
+type PendingBridgeBatch struct {
+	*contractsapi.BridgeMessageBatch
+	Epoch uint64
+}
+
+// NewPendingBridgeBatch creates a new PendingBridgeBatch object
+func NewPendingBridgeBatch(epoch uint64,
+	bridgeEvents []*contractsapi.BridgeMsgEvent) (*PendingBridgeBatch, error) {
+	messages := make([]*contractsapi.BridgeMessage, len(bridgeEvents))
+
+	for i, bridgeEvent := range bridgeEvents {
+		messages[i] = &contractsapi.BridgeMessage{
+			ID:                 bridgeEvent.ID,
+			Sender:             bridgeEvent.Sender,
+			Receiver:           bridgeEvent.Receiver,
+			Payload:            bridgeEvent.Data,
+			SourceChainID:      bridgeEvent.SourceChainID,
+			DestinationChainID: bridgeEvent.DestinationChainID}
+	}
+
+	return &PendingBridgeBatch{
+		BridgeMessageBatch: &contractsapi.BridgeMessageBatch{
+			Messages:           messages,
+			DestinationChainID: bridgeEvents[0].DestinationChainID,
+			SourceChainID:      bridgeEvents[0].SourceChainID},
+		Epoch: epoch,
+	}, nil
```

Comment:
I'd suggest the following changes here:

```suggestion
	if len(bridgeEvents) == 0 {
		return nil, nil // Should we return an error maybe?
	}

	messages := make([]*contractsapi.BridgeMessage, len(bridgeEvents))

	var (
		sourceChainID, destinationChainID *big.Int
	)

	for i, bridgeEvent := range bridgeEvents {
		if i == 0 {
			sourceChainID = bridgeEvent.SourceChainID
			destinationChainID = bridgeEvent.DestinationChainID
		}

		messages[i] = &contractsapi.BridgeMessage{
			ID:                 bridgeEvent.ID,
			Sender:             bridgeEvent.Sender,
			Receiver:           bridgeEvent.Receiver,
			Payload:            bridgeEvent.Data,
			SourceChainID:      bridgeEvent.SourceChainID,
			DestinationChainID: bridgeEvent.DestinationChainID}
	}

	return &PendingBridgeBatch{
		BridgeMessageBatch: &contractsapi.BridgeMessageBatch{
			Messages:           messages,
			DestinationChainID: destinationChainID,
			SourceChainID:      sourceChainID},
		Epoch: epoch,
	}, nil
```